lib/fileset/default.nix

```diff
@@ -286,83 +329,98 @@ in {
 
     The name of the store path is always `source`.
 
-    Type:
-      toSource :: {
-        root :: Path,
-        fileset :: FileSet,
-      } -> SourceLike
-
-    Example:
-      # Import the current directory into the store
-      # but only include files under ./src
-      toSource {
-        root = ./.;
-        fileset = ./src;
-      }
-      => "/nix/store/...-source"
-
-      # Import the current directory into the store
-      # but only include ./Makefile and all files under ./src
-      toSource {
-        root = ./.;
-        fileset = union
-          ./Makefile
-          ./src;
-      }
-      => "/nix/store/...-source"
-
-      # Trying to include a file outside the root will fail
-      toSource {
-        root = ./.;
-        fileset = unions [
-          ./Makefile
-          ./src
-          ../LICENSE
-        ];
-      }
-      => <error>
-
-      # The root needs to point to a directory that contains all the files
-      toSource {
-        root = ../.;
-        fileset = unions [
-          ./Makefile
-          ./src
-          ../LICENSE
-        ];
-      }
-      => "/nix/store/...-source"
-
-      # The root has to be a local filesystem path
-      toSource {
-        root = "/nix/store/...-source";
-        fileset = ./.;
-      }
-      => <error>
+    # Inputs
+
+    Takes an attribute set with the following attributes
+
+    ## root (required, Path)
```

Comment:
Ideally this shouldn't be a heading, but instead continue to be a definition list just like with other functions. Is there a reason why you wanted to turn them into headings?

Reply:
I wanted to try a better way than doing a nested definition list. Because this function only takes on input which is an attribute set with some attributes.

This would be

```md

structured function argument

: root
    : definition of root
    where must the next line start ?
: fileset
    : definition of fileset
```


I always mess up the indentation with nested definition lists-

Reply:
In this particular case, you won't need a nested definition list though, right? You could convert the heading style into a single definition list:

```
`root` (Path; _required_)
: definition of root

`fileset` (FileSet; _required_)
: definition of fileset
```

As for the indentation, I always follow the same [indentation rule](https://spec.commonmark.org/0.31.2/#list-items) as normal markdown lists and nested lists: make sure the content of the definition starts at the same level of indentation:

```
`attributeSet`
: Definition of attribute set.
  Note that it starts in column 2 of the text (counting from 0), so any lines that follow it and should be part of the definition must also have at least 2 spaces at the beginning.

  Line breaks are ok, as long as the content still has the 2 spaces at the beginning.

  `attribute1`
  : This is a definition for an attribute of `attributeSet`.
    Note that it starts in column 4 now, so any lines that should be part of `attribute` must also have at least 4 spaces at the beginning.

    As always, line breaks are still ok, as long as the content has 4 spaces at the beginning.

  `attribute2`
  : Since `attribute2` started with only 2 spaces of indentation, it's part of the content of `attributeSet` instead of the content for `attribute1`.
```

![20240417T043740](https://github.com/NixOS/nixpkgs/assets/160084/b6aab902-963e-45fd-b761-b76d311bdf04)

You can't go wrong by ensuring content always starts on the same column.